scvi/lightning/_vaetask.py

```diff
@@ -0,0 +1,487 @@
+from inspect import getfullargspec
+from typing import Union
+
+import pytorch_lightning as pl
+import torch
+from torch.nn import functional as F
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+
+from scvi import _CONSTANTS
+from scvi._compat import Literal
+from scvi.compose import AbstractVAE, one_hot
+from scvi.modules import Classifier
+
+
+class VAETask(pl.LightningModule):
+    """
+    Lightning module task to train scvi-tools modules
+
+    Parameters
+    ----------
+    vae_model
+        A model instance from class ``AbstractVAE``.
```

Comment:
For terminology, that's not really a model. Let's be careful to keep this in mind

Reply:
It's confusing because of what we call model (the high-level API)